letsencrypt/models/base_config_settings.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+from inspect import getmembers, isfunction
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    letsencrypt_dns_provider = fields.Selection(
+        selection='_get_supported_dns_providers')
+    letsencrypt_dns_login = fields.Char()
+    letsencrypt_dns_key = fields.Char()
+    letsencrypt_dns_url = fields.Char()
+    letsencrypt_needs_dns_provider = fields.Boolean()
+
+    def _get_supported_dns_providers(self):
```

Comment:
how is this better than having the list being produced by a function and every addon adding a provider just adds its value to the list?

the function names become interesting when *actually doing something*, there we need some protocol to call a function based on the provider

Reply:
1) I was under the impression that everyone that would want to add a new provider would add it under this module, but apparently other modules that would be created will depend on this one?
2) If I do that then why do I even need a function? Just a Selection field with a single entry (the first one) and all the others can use `selection_add`?

Reply:
1. yes. It's very probable that every of those providers comes with an extra dependency, so adding all of this to one module would include a lot of stuff most users won't ever need. Also some providers are only interesting in some countries, so those are better off in the appropriate l10n repo
2. just having every addon adding to the selection list is perfectly fine